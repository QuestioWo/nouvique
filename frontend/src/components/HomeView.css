.subtitle{
  font-family: 'Sanchez', serif;
  text-align: center;
  padding-top: 1rem;
  padding-bottom:2.5rem;
}

.content{
  padding-top: 2.5rem;
  background-color:rgb(232, 190, 144) ;
}