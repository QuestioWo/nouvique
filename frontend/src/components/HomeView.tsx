import React from 'react';

import { Result } from 'neverthrow';
import { Row } from 'react-bootstrap';
import { RouteComponentProps } from 'react-router';

import { ApiTest, apiTestLink, PageProps, resolveGETCall } from '../utils';

import BasePage from './elements/BasePage';

import './HomeView.css';

interface Props extends RouteComponentProps, PageProps { }

interface State {
	apiValue: ApiTest
}

export default class HomeView extends React.Component<Props, State> {
	constructor(props: Props) {
		super(props);

		this.state = {
			apiValue: {
				blah: ""
			}
		};
	}

	async componentDidMount() {
		const result: Result<ApiTest, Error> = await resolveGETCall<ApiTest>(apiTestLink);

		result
			.map(res => {
				this.setState({ apiValue: res });

				return null; // necessary to silence warning
			})
			.mapErr(err => {
				console.error(err);
			});
	}

	render() {
		return (
			<React.Fragment>
				<BasePage {...this.props}>
					<Row>
						<h1>
							Home
						</h1>
					</Row>

					<Row>
						<div>
							{this.state.apiValue.blah}
						</div>
					</Row>
				</BasePage>
			</React.Fragment>
		);
	}
}