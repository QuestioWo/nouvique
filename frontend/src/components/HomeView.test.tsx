import * as React from 'react';

import { mount, shallow } from 'enzyme';

import { HomeView, HomeViewProps } from './HomeView';
import { MemoryRouter as Router } from 'react-router';
import { loadStripe } from '@stripe/stripe-js';

describe('HomeView', () => {
	const getProps = (): HomeViewProps => (
		{
			history: {} as any,
			location: {} as any,
			match: {} as any,
			updateAlertBar: async () => { },
			emptyBasket: () => { },
			addToBasket: () => { },
			removeFromBasket: () => { },
			getBasketItems: () => [],
			getTotalBasketPrice: () => 0,
			checkBasketAvailabilities: async () => { },
			stripePromise: loadStripe("foo"),
			alert: {} as any
		}
	)

	const getComponent = (props: HomeViewProps) => (
		<HomeView {...props} />
	);


	it('Should render correctly with shallow', () => {
		const component = shallow(
			getComponent(getProps())
		);
		expect(component.debug()).toMatchSnapshot();
	});

	it('Should render correctly with mount', () => {
		// have to include router as will fail with react-router problem due to use of specific functions
		const component = mount(
			<Router>
				{getComponent(getProps())}
			</Router>
		);
		expect(component.debug()).toMatchSnapshot();
	});
});