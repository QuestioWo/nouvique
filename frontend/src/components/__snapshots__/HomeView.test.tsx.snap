// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView Should render correctly with mount 1`] = `
"<MemoryRouter>
  <Router history={{...}}>
    <HomeView history={{...}} location={{...}} match={{...}} updateAlertBar={[Function: updateAlertBar]} emptyBasket={[Function: emptyBasket]} addToBasket={[Function: addToBasket]} removeFromBasket={[Function: removeFromBasket]} getBasketItems={[Function: getBasketItems]} getTotalBasketPrice={[Function: getTotalBasketPrice]} checkBasketAvailabilities={[Function: checkBasketAvailabilities]} stripePromise={{...}} alert={{...}}>
      <BasePage history={{...}} location={{...}} match={{...}} updateAlertBar={[Function: updateAlertBar]} emptyBasket={[Function: emptyBasket]} addToBasket={[Function: addToBasket]} removeFromBasket={[Function: removeFromBasket]} getBasketItems={[Function: getBasketItems]} getTotalBasketPrice={[Function: getTotalBasketPrice]} checkBasketAvailabilities={[Function: checkBasketAvailabilities]} stripePromise={{...}} alert={{...}}>
        <div className=\\"master\\">
          <img alt=\\"\\" src=\\"/logo_main.png\\" className=\\"logo\\" />
           
          <NavBar>
            <Navbar bg=\\"transparent\\" variant=\\"light\\" expand={true} collapseOnSelect={false}>
              <nav className=\\"navbar navbar-expand navbar-light bg-transparent\\">
                <NavbarCollapse className=\\"justify-content-center\\">
                  <ForwardRef in={false} className=\\"justify-content-center\\" timeout={300} mountOnEnter={false} unmountOnExit={false} appear={false} getDimensionValue={[Function: getDefaultDimensionValue]}>
                    <ForwardRef addEndListener={[Function: transitionEndListener]} in={false} timeout={300} mountOnEnter={false} unmountOnExit={false} appear={false} aria-expanded={{...}} onEnter={[Function (anonymous)]} onEntering={[Function (anonymous)]} onEntered={[Function (anonymous)]} onExit={[Function (anonymous)]} onExiting={[Function (anonymous)]} childRef={{...}}>
                      <Transition in={false} timeout={300} mountOnEnter={false} unmountOnExit={false} appear={false} aria-expanded={{...}} onEnter={[Function (anonymous)]} onEntered={[Function (anonymous)]} onEntering={[Function (anonymous)]} onExit={[Function (anonymous)]} onExited={[Function (anonymous)]} onExiting={[Function (anonymous)]} addEndListener={[Function (anonymous)]} nodeRef={{...}} enter={true} exit={true}>
                        <div className=\\"justify-content-center navbar-collapse collapse\\" aria-expanded={{...}}>
                          <Nav className=\\"mr-auto nav_style\\" justify={false} fill={false}>
                            <ForwardRef as=\\"div\\" activeKey={[undefined]} className=\\"mr-auto nav_style navbar-nav\\" onSelect={[Function (anonymous)]}>
                              <div className=\\"mr-auto nav_style navbar-nav\\" onKeyDown={[Function: handleKeyDown]} role={[undefined]}>
                                <NavItem className=\\"item_style\\">
                                  <div className=\\"item_style nav-item\\">
                                    <NavLink className=\\"nav-link\\" isActive={[Function: isActive]} activeClassName=\\"nav-link active\\" to={{...}}>
                                      <Link aria-current=\\"page\\" className=\\"nav-link nav-link active\\" style={{...}} to={{...}}>
                                        <LinkAnchor aria-current=\\"page\\" className=\\"nav-link nav-link active\\" style={{...}} href=\\"/\\" navigate={[Function: navigate]}>
                                          <a aria-current=\\"page\\" className=\\"nav-link nav-link active\\" style={{...}} href=\\"/\\" onClick={[Function: onClick]}>
                                            Home
                                          </a>
                                        </LinkAnchor>
                                      </Link>
                                    </NavLink>
                                  </div>
                                </NavItem>
                                <NavItem className=\\"item_style\\">
                                  <div className=\\"item_style nav-item\\">
                                    <NavLink className=\\"nav-link\\" activeClassName=\\"nav-link active\\" to={{...}}>
                                      <Link aria-current={{...}} className=\\"nav-link\\" style={[undefined]} to={{...}}>
                                        <LinkAnchor aria-current={{...}} className=\\"nav-link\\" style={[undefined]} href=\\"/browse\\" navigate={[Function: navigate]}>
                                          <a aria-current={{...}} className=\\"nav-link\\" style={[undefined]} href=\\"/browse\\" onClick={[Function: onClick]}>
                                            Browse
                                          </a>
                                        </LinkAnchor>
                                      </Link>
                                    </NavLink>
                                  </div>
                                </NavItem>
                                <NavItem className=\\"item_style\\">
                                  <div className=\\"item_style nav-item\\">
                                    <NavLink className=\\"nav-link\\" activeClassName=\\"nav-link active\\" to={{...}}>
                                      <Link aria-current={{...}} className=\\"nav-link\\" style={[undefined]} to={{...}}>
                                        <LinkAnchor aria-current={{...}} className=\\"nav-link\\" style={[undefined]} href=\\"/ethics\\" navigate={[Function: navigate]}>
                                          <a aria-current={{...}} className=\\"nav-link\\" style={[undefined]} href=\\"/ethics\\" onClick={[Function: onClick]}>
                                            Ethics
                                          </a>
                                        </LinkAnchor>
                                      </Link>
                                    </NavLink>
                                  </div>
                                </NavItem>
                                <NavItem className=\\"item_style\\">
                                  <div className=\\"item_style nav-item\\">
                                    <NavLink className=\\"nav-link nav-right-side\\" activeClassName=\\"nav-link active nav-right-side\\" to={{...}}>
                                      <Link aria-current={{...}} className=\\"nav-link nav-right-side\\" style={[undefined]} to={{...}}>
                                        <LinkAnchor aria-current={{...}} className=\\"nav-link nav-right-side\\" style={[undefined]} href=\\"/login\\" navigate={[Function: navigate]}>
                                          <a aria-current={{...}} className=\\"nav-link nav-right-side\\" style={[undefined]} href=\\"/login\\" onClick={[Function: onClick]}>
                                            Login
                                          </a>
                                        </LinkAnchor>
                                      </Link>
                                    </NavLink>
                                  </div>
                                </NavItem>
                              </div>
                            </ForwardRef>
                          </Nav>
                        </div>
                      </Transition>
                    </ForwardRef>
                  </ForwardRef>
                </NavbarCollapse>
              </nav>
            </Navbar>
          </NavBar>
          <div className=\\"mainbody\\">
            <AlertBar history={{...}} location={{...}} match={{...}} updateAlertBar={[Function: updateAlertBar]} emptyBasket={[Function: emptyBasket]} addToBasket={[Function: addToBasket]} removeFromBasket={[Function: removeFromBasket]} getBasketItems={[Function: getBasketItems]} getTotalBasketPrice={[Function: getTotalBasketPrice]} checkBasketAvailabilities={[Function: checkBasketAvailabilities]} stripePromise={{...}} alert={{...}}>
              <Alert show={true} variant={[undefined]} onClose={[Function (anonymous)]} dismissible={true} transition={{...}} closeLabel=\\"Close alert\\">
                <Fade unmountOnExit={true} in={true} timeout={300} mountOnEnter={false} appear={false}>
                  <ForwardRef addEndListener={[Function: transitionEndListener]} unmountOnExit={true} in={true} timeout={300} mountOnEnter={false} appear={false} onEnter={[Function (anonymous)]} childRef={{...}}>
                    <Transition unmountOnExit={true} in={true} timeout={300} mountOnEnter={false} appear={false} onEnter={[Function (anonymous)]} onEntered={[Function (anonymous)]} onEntering={[Function (anonymous)]} onExit={[Function (anonymous)]} onExited={[Function (anonymous)]} onExiting={[Function (anonymous)]} addEndListener={[Function (anonymous)]} nodeRef={{...}} enter={true} exit={true}>
                      <div role=\\"alert\\" className=\\"fade alert alert-dismissible show\\">
                        <CloseButton onClick={[Function (anonymous)]} aria-label=\\"Close alert\\" variant={[undefined]}>
                          <button type=\\"button\\" className=\\"btn-close\\" onClick={[Function (anonymous)]} aria-label=\\"Close alert\\" />
                        </CloseButton>
                      </div>
                    </Transition>
                  </ForwardRef>
                </Fade>
              </Alert>
            </AlertBar>
            <Row>
              <div className=\\"row\\">
                <h2 className=\\"subtitle\\">
                  Connecting you to your community through art.
                  <br />
                  Local arists using sutainable methods, curated by us.
                </h2>
              </div>
            </Row>
          </div>
        </div>
        <footer className=\\"footer py-3 text-black\\">
          <div />
          <Container fluid={false}>
            <div className=\\"container\\">
              <Row>
                <div className=\\"row\\">
                  <Col>
                    <div className=\\"col\\">
                      Nouvique 2021 v
                      0.22.0
                    </div>
                  </Col>
                  <Col>
                    <div className=\\"col\\">
                      <div style={{...}}>
                        <Link to=\\"/support\\">
                          <LinkAnchor href=\\"/support\\" navigate={[Function: navigate]}>
                            <a href=\\"/support\\" onClick={[Function: onClick]}>
                              Support
                            </a>
                          </LinkAnchor>
                        </Link>
                      </div>
                    </div>
                  </Col>
                </div>
              </Row>
            </div>
          </Container>
        </footer>
      </BasePage>
    </HomeView>
  </Router>
</MemoryRouter>"
`;

exports[`HomeView Should render correctly with shallow 1`] = `
"<Fragment>
  <BasePage history={{...}} location={{...}} match={{...}} updateAlertBar={[Function: updateAlertBar]} emptyBasket={[Function: emptyBasket]} addToBasket={[Function: addToBasket]} removeFromBasket={[Function: removeFromBasket]} getBasketItems={[Function: getBasketItems]} getTotalBasketPrice={[Function: getTotalBasketPrice]} checkBasketAvailabilities={[Function: checkBasketAvailabilities]} stripePromise={{...}} alert={{...}}>
    <Row>
      <h2 className=\\"subtitle\\">
        Connecting you to your community through art.
        <br />
        Local arists using sutainable methods, curated by us.
      </h2>
    </Row>
  </BasePage>
</Fragment>"
`;
